clients/fides-js/src/components/notices/NoticeOverlay.tsx

```diff
@@ -141,7 +142,14 @@ const NoticeOverlay: FunctionComponent<OverlayProps> = ({
 
   window.addEventListener("FidesUpdating", (event) => {
     // If GPC is being applied after initialization, we need to update the initial overlay to reflect the new state. This is especially important for Firefox browsers (Gecko) because GPC gets applied rather late due to how it handles queuing the `setTimeout` on the last step of our `initialize` function.
-    setDraftEnabledNoticeKeys(initialEnabledNoticeKeys(event.detail.consent));
+    const { consent } = event.detail;
+    Object.keys(consent).forEach((key) => {
+      const value = consent[key];
+      if (typeof value === "string") {
+        consent[key] = transformUserPreferenceToBoolean(value);
+      }
+    });
+    setDraftEnabledNoticeKeys(initialEnabledNoticeKeys(consent));
```

Comment:
Do we need an equivalent to ensure we handle any custom notices in TCF too?

Reply:
Created a bug ticket for GPC in TCF. Resolving this for now.